File: Assignment/assignment4/source.c
#include <stdio.h>
#include <stdlib.h>

void printMenu();
float opAdd(float A, float B);
float opSub(float A, float B);
float opMul(float A, float B);
float opDiv(float A, float B);
float input();

int main() {
	char command = -1;
	int counter = 0;

	float firstOp = 0;
	float secondOp = 0;

	const char opStr[] = " +-*/";

	printf("-----------------------------------\n"
		"         Last Calculator\n"
		"-----------------------------------\n");

	while (1) {
		printMenu();
		scanf(" %c", &command);

		if (command == '0')
			break;
		else if (command != '1' && command != '2' && command != '3' && command != '4') {
			printf("Wrong command!\n");
			continue;
		}

		if (counter == 0) {
			printf("Input first operand: ");
			firstOp = input();
			printf("Input second operand: ");
			secondOp = input();
		} else {
			printf("%.4f %c ", firstOp, opStr[command - '0']);
			secondOp = input();
		}

		if (command == '1')
			firstOp = opAdd(firstOp, secondOp);
		else if (command == '2')
			firstOp = opSub(firstOp, secondOp);
		else if (command == '3')
			firstOp = opMul(firstOp, secondOp);
		else if (command == '4')
			(secondOp != 0) ? (firstOp = opDiv(firstOp, secondOp)) : printf("Can't divide by zero!\n");

		counter = 1;
		printf("Result is %.4f\n", firstOp);
	}

	printf("Good bye!\n");
}

void printMenu() {
	printf(
		"-----------------------------------\n"
		"1. add\n2. sub\n3. mul\n4. div\n0. exit\n"
		"-----------------------------------\n"
		">>> "
	);
}

float opAdd(float A, float B) {
	return A + B;
}

float opSub(float A, float B) {
	return A - B;
}

float opMul(float A, float B) {
	return A * B;
}

float opDiv(float A, float B) {
	return A / B;
}

float input() {
	char temp[256];
	float ret = 0;

	while (1) {
		scanf(" %s", temp);
		ret = atof(temp);

		if (ret == 0 && temp[0] != '0') {
			printf("Wrong input! Input again: ");
			continue;
		}

		return ret;
	}
}

File: Assignment/assignment2/source.py
def printMenu():
    print '------------------------------'
    print '        Calculator'
    print '------------------------------'
    print '        What to do?'
    print '------------------------------'
    print '1. add'
    print '2. sub'
    print '3. mul'
    print '4. div'
    print '0. quit'
    print '------------------------------'
    print '>>> ',

def opAdd(args):
    ret = 0

    for n in args:
        ret += n

    return ret

def opSub(lhs, rhs):
    return lhs - rhs

def opMul(args):
    ret = 1
    
    for n in args:
        ret *= n

    return ret

def opDiv(lhs, rhs):
    return float(lhs) / float(rhs)

while True:
    printMenu()

    cmd = raw_input()
    
    if cmd == '1':
        print 'input as many as you want (input result to quit)'

        nums = []

        while True:
            line = raw_input()

            if line == 'result':
                break

            nums.append(float(line))

        print '%d numbers added, result is %0.2f' % (len(nums), opAdd(nums))

    elif cmd == '2':
        print 'first oprand : ',
        firstOp = input()
        print 'second oprand : ',
        secondOp = input()

        print (str(firstOp) + ' - ' + str(secondOp) + ' = %0.2f' % opSub(firstOp, secondOp))

    elif cmd == '3':
        print 'input as many as you want (input result to quit)'

        nums = []

        while True:
            line = raw_input()

            if line == 'result':
                break

            nums.append(float(line))

        print '%d numbers multiplied, result is %0.2f' % (len(nums), opMul(nums))

    elif cmd == '4':
        print 'first oprand : ',
        firstOp = input()
        print 'second oprand : ',
        secondOp = input()

        if secondOp == 0:
            print 'Can\'t divide by zero!!!'
        else:
            print(str(firstOp) + '/' + str(secondOp) + ' = %0.2f' % opDiv(firstOp, secondOp))
    elif cmd == '0':
        print 'bye'
        break
    else:
        print 'Wrong command!'
        continue
